src/linear_algebra/contraction.lean

```diff
@@ -51,4 +71,94 @@ variables {R M N}
   dual_tensor_hom R M N (f ⊗ₜ n) m = (f m) • n :=
 by { dunfold dual_tensor_hom, rw uncurry_apply, refl, }
 
+/-- As a matrix, `dual_tensor_hom` evaluated on a basis element of `M* ⊗ N` is a matrix with a
+single one and zeros elsewhere -/
+theorem dual_tensor_hom_basis
+  {m : Type*} {n : Type*} [fintype m] [fintype n] [decidable_eq m] [decidable_eq n]
+  (bM : basis m R M) (bN : basis n R N) (j : m) (i : n) :
+    to_matrix bM bN (dual_tensor_hom R M N (bM.coord j ⊗ₜ bN i)) = std_basis_matrix i j 1 :=
+begin
+  ext i' j',
+  by_cases hij : (i = i' ∧ j = j');
+  simp [linear_map.to_matrix_apply, finsupp.single_eq_pi_single, hij],
+  rw [and_iff_not_or_not, not_not] at hij, cases hij; simp [hij],
+end
+
+/-- `hom_dual_tensor` is right inverse to `dual_tensor_hom` -/
+@[simp]
+lemma dual_tensor_hom_hom_dual_tensor_apply
+  {ι : Type} [fintype ι] [decidable_eq ι] (b : basis ι R M) (f : M →ₗ[R] N):
+  (dual_tensor_hom R M N) ((hom_dual_tensor R M N b) f) = f :=
+begin
+  ext m, nth_rewrite_rhs 0 ←basis.sum_repr b m,
+  simp [hom_dual_tensor],
+end
+
+/-- `hom_dual_tensor` is right inverse to `dual_tensor_hom` -/
+lemma dual_tensor_hom_hom_dual_tensor {ι : Type} [fintype ι] [decidable_eq ι] (b : basis ι R M):
+  (dual_tensor_hom R M N) ∘ₗ (hom_dual_tensor R M N b) = id :=
+by { ext f, simp }
+
+
+/-- `hom_dual_tensor` is left inverse to `dual_tensor_hom` -/
+lemma hom_dual_tensor_dual_tensor_hom {ι : Type} [fintype ι] [decidable_eq ι] (b : basis ι R M):
+  (hom_dual_tensor R M N b) ∘ₗ (dual_tensor_hom R M N) = id :=
+begin
+  apply curry_injective,
+  apply basis.ext b.dual_basis,
+  intro i, ext n,
+  suffices H : ∑ (x : ι), (finsupp.single x (1:R)) i • b.coord x ⊗ₜ[R] n = b.coord i ⊗ₜ[R] n,
+  { simp [hom_dual_tensor_apply, -finsupp.single_one_smul, H] },
+  simp,
+  --rw finsupp.fintype_sum_single,
+  sorry,
```

Comment:
Just checking you saw this comment: https://github.com/leanprover-community/mathlib/pull/10372/files#r753077439